File: Python3.x/1109-Corporate Flight Bookings.py
# Time Limit Exceeded
class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        res = [0] * n
        for i, j, k in bookings:
            for idx in range(i - 1, j):
                res[idx] += k
        return res


class Solution:
    def corpFlightBookings(self, bookings: List[List[int]], n: int) -> List[int]:
        res = [0] * n
        for i, j, k in bookings:
            res[i - 1] += k
            if j < n:
                res[j] -= k
        for i in range(1, n):
            res[i] += res[i - 1]
        return res

File: C++/1109-Corporate_Flight_Bookings.cpp
class Solution {
public:
    vector<int> corpFlightBookings(vector<vector<int>>& bookings, int n) {
        vector<int> res(n);
        for (vector<int> v : bookings) {
            res[v[0] - 1] += v[2];
            if (v[1] < n)
                res[v[1]] -= v[2];
        }
        for (int i = 1; i < n; ++i)
            res[i] += res[i - 1];
        return res;
    }
};
